core/block/object/objectcreator/creator.go

```diff
@@ -311,71 +311,47 @@ func (w *Creator) createRelationOption(ctx context.Context, spaceID string, deta
-	id = key
-	object.Fields[bundle.RelationKeyId.String()] = pbtypes.String(id)
+	object = pbtypes.CopyStruct(details)
+	object.Fields[bundle.RelationKeyUniqueKey.String()] = pbtypes.String(uniqueKey.Marshal())
 	object.Fields[bundle.RelationKeyLayout.String()] = pbtypes.Int64(int64(model.ObjectType_relationOption))
 
-	return w.CreateSmartBlockFromState(ctx, spaceID, coresb.SmartBlockTypeRelation, []domain.TypeKey{bundle.TypeKeyRelationOption}, object, nil)
-}
-
-type internalKeyGetter interface {
-	InternalKey() string
+	createState := state.NewDocWithUniqueKey("", nil, uniqueKey).(*state.State)
+	createState.SetDetails(object)
+	return w.CreateSmartBlockFromState(ctx, spaceID, coresb.SmartBlockTypeRelation, []domain.TypeKey{bundle.TypeKeyRelationOption}, nil, createState)
 }
 
 func (w *Creator) createObjectType(ctx context.Context, spaceID string, details *types.Struct) (id string, newDetails *types.Struct, err error) {
 	if details == nil || details.Fields == nil {
 		return "", nil, fmt.Errorf("create object type: no data")
 	}
 
-	sbType := coresb.SmartBlockTypeObjectType
-	uk, err := getUniqueKeyOrGenerate(sbType, details)
+	uniqueKey, err := getUniqueKeyOrGenerate(coresb.SmartBlockTypeObjectType, details)
 	if err != nil {
-		return "", nil, err
-	}
-	details.Fields[bundle.RelationKeyUniqueKey.String()] = pbtypes.String(uk.Marshal())
-	key := uk.(internalKeyGetter).InternalKey()
-	// TODO Is it ok? Do recommendedRElation contain IDs or keys?
-	recommendedRelationIDs := pbtypes.GetStringList(details, bundle.RelationKeyRecommendedRelations.String())
-	recommendedRelationKeys := make([]string, 0, len(recommendedRelationIDs))
-	for _, relId := range recommendedRelationIDs {
-		relKey, err2 := pbtypes.BundledRelationIdToKey(relId)
-		if err2 != nil {
-			log.Errorf("create object type: invalid recommended relation id: %s", relId)
-			continue
-		}
-		// todo: support custom relations here
-		rel, _ := bundle.GetRelation(domain.RelationKey(relKey))
-		if rel != nil {
-			_, _, err2 := w.createRelation(ctx, spaceID, (&relationutils.Relation{rel}).ToStruct())
-			// todo: check if the relation already exists
-			if err2 != nil && err2 != fmt.Errorf("TODO: relation already exists") {
-				err = fmt.Errorf("failed to create relation for objectType: %s", err2.Error())
-				return
-			}
-		}
-		recommendedRelationKeys = append(recommendedRelationKeys, relKey)
+		return "", nil, fmt.Errorf("getUniqueKeyOrGenerate: %w", err)
 	}
+	details.Fields[bundle.RelationKeyUniqueKey.String()] = pbtypes.String(uniqueKey.Marshal())
+	key := uniqueKey.InternalKey()
+
 	object := pbtypes.CopyStruct(details)
 	rawRecommendedLayout := pbtypes.GetInt64(details, bundle.RelationKeyRecommendedLayout.String())
 	recommendedLayout, err := bundle.GetLayout(model.ObjectTypeLayout(int32(rawRecommendedLayout)))
 	if err != nil {
 		return "", nil, fmt.Errorf("invalid recommended layout %d: %w", rawRecommendedLayout, err)
 	}
 
+	var recommendedRelationKeys []string
```

Comment:
Do we add recommended relations only from layout, but not from recommendedRelations detail as well?

Reply:
I discussed it with Razor, clients don't pass recommendedReleations detail, so I decided to remove this code